src/utils/proton.js

```diff
@@ -37,34 +57,40 @@ class ProtonSDK {
     } catch (e) {
       return e;
     }
-  }
+  };
 
   logout = async () => {
     await this.link.removeSession(this.appName, this.session.auth);
     localStorage.removeItem('savedUserAuth-foobar');
-  }
+  };
 
   restoreSession = async () => {
-    const savedUserAuth = JSON.parse(localStorage.getItem('savedUserAuth-foobar'));
+    const savedUserAuth = JSON.parse(
+      localStorage.getItem('savedUserAuth-foobar')
+    );
     if (savedUserAuth) {
       try {
-        this.link = await ConnectWallet({
-          linkOptions: { chainId: this.chainId, endpoints: this.endpoints },
-          transportOptions: { requestAccount: this.requestAccount, backButton: true },
-          selectorOptions: { appName: this.appName, appLogo: Foobar, showSelector: false }
-        });
-        const result = await this.link.restoreSession(this.requestAccount, savedUserAuth);
-        if (result) {
-          this.session = result;
-          return { auth: this.session.auth, accountData: this.session.accountData[0] };
+        await this.connect(true, false);
+        if (this.session) {
+          const { auth, accountData } = this.session;
+          return {
+            auth,
+            accountData: accountData[0],
+          };
         }
-      } catch(e) {
+      } catch (e) {
         return e;
       }
     }
-    return { auth: { actor: '', permission: '' }, accountData: {}};
-  }
+    return {
+      auth: {
+        actor: '',
+        permission: ''
+      },
+      accountData: {}
+    };
+  };
 }
 
 const protonSDK = new ProtonSDK();
```

Comment:
this can just be `export default new ProtonSDK();`

everything else looks good!